sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1774,6 +1774,19 @@ class CastSuite extends CastSuiteBase {
       assert(e3.contains("Casting 2147483648 to int causes overflow"))
     }
   }
+
+  test("SPARK-35111: Cast string to year-month interval") {
```

Comment:
Could you add round trip tests: string -> year-month interval -> string, and year-month interval -> string -> year-month interval

Reply:
Could you test corner cases when arithmetic overflow happens. Also test upper and lower cases.

Reply:
> Could you test corner cases when arithmetic overflow happens. Also test upper and lower cases.

Current test is not correct according to https://github.com/apache/spark/pull/32281, will update after https://github.com/apache/spark/pull/32281 merged

Reply:
Please, check the input in lower case.